src/clients/localnet.rs

```diff
@@ -0,0 +1,196 @@
+// Solana localnet client for testing against local Solana test validator
+use crate::solana::{
+    base_client::SolanaBaseClient,
+    rpc_client::SolanaRpcClient,
+    signing_client::{SolanaSigningClient, SolanaSigningClientImpl},
+};
+use async_trait::async_trait;
+use solana_sdk::{
+    commitment_config::CommitmentConfig,
+    signature::Keypair,
+};
+
+/// Solana localnet client for development and testing
+pub struct SolanaLocalnetClient {
+    inner: SolanaSigningClientImpl,
+}
+
+impl SolanaLocalnetClient {
```

Comment:
other domain clients mostly rely on initialization based on a mnemonic (str) to keep the calling methods/scope clean of any domain-specific signing logic. could we add a mnemonic-based constructor here as well?